From 7cfdb2d7a314b1c05071a0af3cb0cb5ee9378f80 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Fri, 24 Nov 2017 18:57:29 +0800
Subject: [PATCH 2897/5242] MLK-16977-3 media: camera: refine the reset
 sequence for camera

commit  734d92eb347c87c80842842dcdea5604e60d882e from
https://source.codeaurora.org/external/imx/linux-imx.git

adjust the init sequence for camera thus no rework will be required
for DVDD_1V5.

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../media/platform/mxc/capture/ov5640_mipi_v2.c    |    8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c b/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c
index 58b3320..2ac8478 100644
--- a/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c
+++ b/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c
@@ -472,16 +472,14 @@ static void ov5640_reset(void)
 	gpio_set_value(pwn_gpio, 1);
 	msleep(5);
 
-	gpio_set_value(pwn_gpio, 0);
-	msleep(5);
-
 	gpio_set_value(rst_gpio, 0);
 	msleep(1);
 
-	gpio_set_value(rst_gpio, 1);
+	gpio_set_value(pwn_gpio, 0);
 	msleep(5);
 
-	gpio_set_value(pwn_gpio, 1);
+	gpio_set_value(rst_gpio, 1);
+	msleep(5);
 }
 
 static int ov5640_regulator_enable(struct device *dev)
-- 
1.7.9.5

